storage/deliveryjobrepo.go

```diff
@@ -97,6 +97,40 @@ func (djRepo *DeliveryJobDBRepository) MarkJobRetry(deliveryJob *data.DeliveryJo
 	return err
 }
 
+// MarkDeadJobAsInflight increases the retry attempt count and sets the status of the job to Queued if the job's current status is Dead in the object and DB; else returns error
+func (djRepo *DeliveryJobDBRepository) MarkDeadJobAsInflight(deliveryJob *data.DeliveryJob) (err error) {
+	currentTime := time.Now()
+	err = transactionalSingleRowWriteExec(djRepo.db, emptyOps, "UPDATE job SET status = ?, statusChangedAt = ?, updatedAt = ?, earliestNextAttemptAt = ?, retryAttemptCount = ? WHERE id like ? and status = ?", args2SliceFnWrapper(data.JobInflight, currentTime, currentTime, currentTime, deliveryJob.RetryAttemptCount+1, deliveryJob.ID, data.JobDead))
+	if err == nil {
+		deliveryJob.Status = data.JobInflight
+		deliveryJob.StatusChangedAt = currentTime
+		deliveryJob.UpdatedAt = currentTime
+		deliveryJob.EarliestNextAttemptAt = currentTime
+		deliveryJob.RetryAttemptCount = deliveryJob.RetryAttemptCount + 1
+	}
+	return err
+}
+
+func (djRepo *DeliveryJobDBRepository) populateJobsWithMessage(jobs []*data.DeliveryJob) error {
+	messageIDs := make([]string, 0, len(jobs))
+	for _, job := range jobs {
+		messageIDs = append(messageIDs, job.Message.ID.String())
+	}
+	messages, err := djRepo.mesageRepository.GetByIDs(messageIDs)
+	if err == nil {
+		messageMap := make(map[string]*data.Message)
+		for _, message := range messages {
+			messageMap[message.ID.String()] = message
+		}
+		for _, job := range jobs {
+			if err == nil {
```

Comment:
Is this check necessary?
We aren't setting the `err` inside of the of `if err == nil` block

Reply:
Right 😢, forgot to remove this after moving code.

Reply:
Removed check.